ddsenabler/examples/action/CLIParser.hpp

```diff
@@ -0,0 +1,331 @@
+// Copyright 2025 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <csignal>
+#include <cstdlib>
+#include <filesystem>
+#include <iostream>
+
+#include <fastdds/dds/log/Log.hpp>
+
+#pragma once
+
+class CLIParser
+{
+public:
+
+    CLIParser() = delete;
+
+    struct example_config
+    {
+        std::string config_file_path = "";
+        std::string action_name = "fibonacci/_action/";
+        bool announce_server = true;
+        uint32_t timeout = 30;
+        std::string persistence_path = "";
+        std::string goals_path = "";
+        uint32_t expected_requests = 0;
+        uint32_t request_initial_wait = 0;
+        bool cancel_requests = false;
+    };
+
+    /**
+     * @brief Print usage help message and exit with the given return code
+     *
+     * @param return_code return code to exit with
+     *
+     * @warning This method finishes the execution of the program with the input return code
+     */
+    static void print_help(
+            uint8_t return_code)
+    {
+        std::cout << "Usage: ddsenabler_example_action <client|server> [options]"                                 <<
```

Comment:
NIT: please polish helpers (review also service example); check lines are aligned and all defaults are specified.